libbpf-rs/src/object.rs

```diff
@@ -167,12 +165,58 @@ pub struct OpenObject {
 }
 
 impl OpenObject {
-    fn new(ptr: *mut libbpf_sys::bpf_object) -> Self {
-        OpenObject {
+    fn new(ptr: *mut libbpf_sys::bpf_object) -> Result<Self> {
+        let mut obj = OpenObject {
             ptr,
             maps: HashMap::new(),
             progs: HashMap::new(),
+        };
+
+        // Populate obj.maps
+        let mut map: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+        loop {
+            // Get the pointer to the next BPF map
+            let next_ptr = unsafe { libbpf_sys::bpf_map__next(map, obj.ptr) };
+            if next_ptr.is_null() {
+                break;
+            }
+
+            // Get the map name
+            let name = unsafe { libbpf_sys::bpf_map__name(next_ptr) };
+            let err = unsafe { libbpf_sys::libbpf_get_error(name as *const _) };
+            if err != 0 {
+                return Err(Error::System(err as i32));
+            }
+            let name = util::c_ptr_to_string(name)?;
+
+            // Add the map to the hashmap
+            obj.maps.insert(name, OpenMap::new(next_ptr));
+            map = next_ptr;
+        }
+
+        // Populate obj.progs
+        let mut prog: *mut libbpf_sys::bpf_program = std::ptr::null_mut();
+        loop {
+            // Get the pointer to the next BPF program
+            let next_ptr = unsafe { libbpf_sys::bpf_program__next(prog, obj.ptr) };
+            if next_ptr.is_null() {
+                break;
+            }
+
+            // Get the program name
+            let name = unsafe { libbpf_sys::bpf_program__name(next_ptr) };
```

Comment:
it's even stricter for bpf_program__name, it doesn't expect NULL and never returns NULL

Reply:
Fixed in latest commit